staging/src/k8s.io/apiserver/pkg/util/webhook/authentication.go

```diff
@@ -46,7 +49,23 @@ func NewDefaultAuthenticationInfoResolverWrapper(
 				if hostPort == "kubernetes.default.svc:443" {
 					return kubeapiserverClientConfig, nil
 				}
-				return delegate.ClientConfigFor(hostPort)
+				ret, err := delegate.ClientConfigFor(hostPort)
+				if err != nil {
+					return nil, err
+				}
+
+				if egressLookup != nil {
+					networkContext := egressselector.Master.AsNetworkContext()
```

Comment:
I think this makes sense because this traffic is for non-services and those should be hosted similar to authenticator and authorization webhooks.

cc @cheftako for a second opinion.

Reply:
Agreed